### DISCUSSION TITLE
DNS resolution issue

### DISCUSSION THREAD

**Discussion by ibrahimk157** [angristan/wireguard-install#429] [Help]
Hello,

I recently installed wireguard using this script and for some odd reason it broke the server's DNS (it would fail to resolve) and when temporarily setting the DNS servers in /etc/resolv.conf which fixed the problem temporarily (I have to repeat this after every reboot). Mind you at this time, when connecting to the wireguard server on the target clients/devices, everything worked just fine.

So I tried several guides online to permanently set the server's DNS and none worked (can't even remember which guides I've used). So I went ahead and un-did everything. However, I was then presented with another issue. 

Now the clients connected to the wireguard server fail to resolve the DNS (visiting websites with domain names won't work, but visiting direct IPs worked fine). And when disconnecting them from the wireguard server everything goes back to working perfectly. 

I double checked the wireguard client config files and the DNS servers (in my case adguard) are all correctly there.

Tried changing the MTU value as some posts online suggested but that didn't work either.

I also tried uninstalling wireguard using this script and then re-installing it and configuring new clients with new profiles, yet the same issue occurred.

I genuinely do not know what to do now. Can anyone mind helping me with this? I'm more than happy to provide any logs (just please let me know how I can obtain these).

Here's the setup of the server:
OS: Ubuntu 22.04 LTS
Server type: Interserver KVM VPS
Network interface: ens3
Has both IPv4 and IPv6

Thanks

**Comment by x-FBi** [comment]
https://www.tecmint.com/set-permanent-dns-nameservers-in-ubuntu-debian/
Set permanent DNS nameservers

**Reply by ibrahimk157** [reply]
Hi!

sorry for the late response.

Tried that and while it fixed the dns resolution within the server, the clients still can’t resolve dns when connected to the server via wireguard.

and it’s quite weird as it would sometimes start working by itself while other times it breaks randomly, despite me not touching anything in the server. I have already tried everything I could find online so I am quite frustrated honestly as nothing seems to work. Even tried tailscale (which uses wireguard) and the same thing happened and even their support team wasn’t able to figure it out either.

**Reply by Merlincool** [reply]
> https://www.tecmint.com/set-permanent-dns-nameservers-in-ubuntu-debian/ Set permanent DNS nameservers

Doesn't work on Debian 10
